Adapt to recent version of PHPUnit (10)

DavidPrevot opened the pull request:
Those changes allow the testsuite to also pass on PHPUnit 10.

Regards,

David

Commit:
Adapt to recent version of PHPUnit (10)

greg0ire commented:
Hi David! It looks like you forgot to bump the version constraint here: https://github.com/doctrine/instantiator/blob/d6eef505a6c46e963e54bf73bb9de43cdea70821/composer.json#L28

In case you're wondering, this is why the checks are red.

Please make the change, amend your commit and force push.

DavidPrevot commented:
Thanks for the hint (I thought I actually run the testsuite with PHPUnit 9 after the change, but obviously forgot).

greg0ire commented:
You're welcome. I believe these lines need to be changed as well: https://github.com/doctrine/instantiator/blob/2.0.x/phpunit.xml.dist#L22-L26

They use old vocabulary (`filter`/ `whitelist` vs `coverage` / `include`)

Commit:
Migrate PHPUnit configuration file to current format

Manually apply the “phpunit --migrate-configuration” changes to keep the
previous layout, and also drop the xsi:noNamespaceSchemaLocation change.

greg0ire commented on phpunit.xml.dist:
```diff
@@ -1,27 +1,15 @@
-            <directory suffix=".php">./src</directory>
-        </whitelist>
-    </filter>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd" bootstrap="./vendor/autoload.php" colors="true" stopOnFailure="false" processIsolation="false" backupGlobals="false" cacheDirectory=".phpunit.cache">
```
Can you please use linebreaks here? Like in the previous version?

greg0ire commented on phpunit.xml.dist:
```diff
@@ -1,27 +1,23 @@
 <?xml version="1.0"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+    xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/10.1/phpunit.xsd"
```
Please revert this change, IMO it's more guaranteed to be the right version of the xsd file here.

greg0ire commented:
Thanks @DavidPrevot !
